lib/raxx/context.ex

```diff
@@ -0,0 +1,110 @@
+defmodule Raxx.Context do
+  @type section_name :: term()
+
+  @typedoc """
+  An opaque type for the context snapshot data.
+  """
+  @opaque snapshot :: map()
+
+  @moduledoc """
+  `Raxx.Context` is a mechanism for simple sharing of state/information between
+  `Raxx.Middleware`s and `Raxx.Server`s.
+
+  It is designed to be flexible and to enable different middlewares to operate
+  on it without conflicts. Each separate functionality using the context
+  can be in a different "section", containing arbitrary data.
+
+  Context is implicitly shared using the process dictionary and persists for the
+  duration of a single request/response cycle. If you want to pass the context
+  to a different process, you need to take its snapshot, pass it explicitly and
+  "restore" it in the other process. See `Raxx.Context.get_snapshot/0` and 
+  `Raxx.Context.restore_snapshot/1` for details.
+  """
+
+  @doc """
+  Sets the value of a context section.
+
+  Returns the previous value of the section or `nil` if one was
+  not set.
+  """
+  @spec set(section_name, term) :: term | nil
+  def set(section_name, value) do
```

Comment:
`section_name` -> `key`?

Reply:
If we're keeping the "section" concept (which I would like), I actually like `section_name` here. I think it drives home the point that a section consists of a name and a value, both of which can be of whatever type.

Reply:
Fair enough I just wanted to check was deliberate. I think in the tag function at some point the same thing is called `key`

Reply:
Yeah, the `tag` function is very private, utilitarian and not-opinionated, I think it makes sense for the naming there to be less informative.